### ISSUE TITLE
Potential race condition when SVEAManager gets state from LocalizationInterface

### ISSUE THREAD

**Issue by rcywongaa** [KTH-SML/svea#32]
In https://github.com/KTH-SML/svea/blob/main/src/svea_core/src/svea/svea_managers/svea_archetypes.py#L68

`SVEAManager` directly gets the current state from the `LocalizationInterface`.

However, as per https://github.com/KTH-SML/svea/blob/main/src/svea_core/src/svea/interfaces/localization.py#L90
`state` is updated in a separate thread.

This could lead to a race condition

**Comment by kaarmu** [comment]
You are correct and this is something we are aware of. However, I'm surprised if you actually encountered a problem. Usually, the manager's state is only used for reading, meaning that race conditions would only cause partial updates to the state. Generally this hasn't been a big problem. Perhaps you encountered something in simulation? I know that there can arise funny effects (that are hard to debug!) between the simulator and localization interface.

**Comment by kaarmu** [comment]
Resolved in ROS 2 updates, #55 and #60
